src/mesh/data/gridfromfile.cxx

```diff
@@ -166,27 +166,62 @@ bool GridFile::get(Mesh *m, Field2D &var,   const string &name, BoutReal def) {
   int ys = m->OffsetY;
 
   // Index offsets into destination
-  int xd = 0;
-  int yd = m->ystart;
-
-  // Number of points to read
-  int nx = m->LocalNx;
-  int ny = m->yend - m->ystart + 1;
-  
-  for(int x=xs;x < xs+nx; x++) {
+  int xd = -1;
+  int yd = -1;
+
+  ///Ghost region width. This is error prone. Should check if really integer!!
```

Comment:
Maybe could use "/2" rather than "0.5*" so it uses integer arithmetic rather than floating point (where rounding may do strange things). You could also have a check before, something like:

ASSERT1((m->LocalNx - (m->xend - m->xstart + 1)) % 2 == 0);

to check that it's a multiple of 2